tests/e2e/scale/upgrade/test_upgrade_with_scaled_rgw_obc.py

```diff
@@ -0,0 +1,186 @@
+        validate=True,
+    )
+    # Validate objects in bucket(s) after created
+    scale_noobaa_lib.validate_bucket(
+        num_objs,
+        upgrade="pre_upgrade",
+        result="result.csv",
+        put=True,
+        get=True,
+        list_obj=True,
+    )
+
+    # Create OBC without I/O and ensure OBC in Bound state before upgrade
+    obc_scaled_list = []
+    log.info(f"Start creating  {scale_obc_count} " f"OBC in a batch of {num_obc_batch}")
+    for i in range(int(scale_obc_count / num_obc_batch)):
+        obc_dict_list = scale_noobaa_lib.construct_obc_creation_yaml_bulk_for_kube_job(
+            no_of_obc=num_obc_batch,
+            sc_name=sc_name,
+            namespace=namespace,
+        )
+        # Create job profile
+        job_file = ObjectConfFile(
+            name="job_profile",
+            obj_dict_list=obc_dict_list,
+            project=namespace,
+            tmp_path=tmp_path,
+        )
+        # Create kube_job
+        job_file.create(namespace=namespace)
+        time.sleep(timeout * 5)
+
+        # Check all the OBCs reached Bound state
+        obc_bound_list = scale_noobaa_lib.check_all_obc_reached_bound_state_in_kube_job(
+            kube_job_obj=job_file,
+            namespace=namespace,
+            no_of_obc=num_obc_batch,
+        )
+        obc_scaled_list.extend(obc_bound_list)
+
+    log.info(
+        f"Number of OBCs in scaled list: {len(obc_scaled_list)}",
+    )
+
+    # Write namespace, OBC data in a OBC_SCALE_DATA_FILE which
+    # will be used during post_upgrade validation tests
+    with open(obc_scaled_data_file, "a+") as w_obj:
+        w_obj.write(str("# Scale Data File\n"))
+        w_obj.write(str(f"NAMESPACE: {namespace}\n"))
+        w_obj.write(str(f"OBC_SCALE_LIST: {obc_scaled_list}\n"))
+
+
+@post_upgrade
+@vsphere_platform_required
+@skipif_external_mode
+@skipif_bm
+@skipif_managed_service
+@pytest.mark.polarion_id("OCS-3988")
+def test_scale_obc_rgw_post_upgrade():
+    """
+    Validate OBC scaled for post upgrade
+    """
+
+    # Get info from SCALE_DATA_FILE for validation
+    if os.path.exists(obc_scaled_data_file):
+        file_data = templating.load_yaml(obc_scaled_data_file)
+        namespace = file_data.get("NAMESPACE")
+        obc_scale_list = file_data.get("OBC_SCALE_LIST")
+    else:
+        raise FileNotFoundError
+
+    # Check obc status in current namespace
+    obc_bound_list, obc_not_bound_list = scale_noobaa_lib.check_all_obcs_status(
+        namespace
+    )
+
+    # Check status of OBC scaled in pre-upgrade
+    if not len(obc_bound_list) == len(obc_scale_list):
+        raise UnexpectedBehaviour(
+            f" OBC bound list count mismatch {len(obc_not_bound_list)} OBCs not in Bound state "
+            f" OBCs not in Bound state {obc_not_bound_list}"
+        )
+    else:
+        log.info(f" Expected all {len(obc_bound_list)} OBCs are in Bound state")
+
+    # Validate existing objects in bucket(s)
+    scale_noobaa_lib.validate_bucket(
```

Comment:
I'm not sure how these test works during upgrade! but I'm curious how the hsbench() instance from pre_upgrade test will be valid in this post_upgrade test?

Reply:
The purpose of this test case is testing on existing objects which are created before the upgrade still function after software upgrade complete. For post upgrade, these operations below are still valid:
* Old objects should be readable/get/list 
* Compare pre and post upgrade objects content in bucket 
* Old Objects should be modify/delete 
* New objects can be created and readable/modify/delete/list